src/workerd/io/worker.c++

```diff
@@ -1572,6 +1589,14 @@ Worker::Worker(kj::Own<const Script> scriptParam,
               }
             }
 
+            KJ_IF_SOME(s, startupTime) {
+              KJ_SWITCH_ONEOF(limitErrorOrTime) {
+                KJ_CASE_ONEOF(startupTimeElapsed, kj::Duration) {
+                  s = startupTimeElapsed;
+                }
+                KJ_CASE_ONEOF_DEFAULT { }
+              }
+            }
```

Comment:
It seems this is the only place where the duration part of the `limitErrorOrTime` is used. It makes me wonder whether a separate argument for this wouldn't be better, unless I'm missing something it would avoid a lot of the refactoring that you needed to do here (which largely is just changing from `KJ_IF_SOME` to `KJ_CASE_ONEOF`).

Reply:
I thought about it in the beginning but I decided against it.
Having a OneOf is clearer in tems of what to expect from enterStartup*.
The result will be EITHER an time exceeded error or the time elapsed, never both, never neither.
I realize it required a bit of refactoring to do this change but it makes a lot more sense semantically.

Reply:
Ahh, didn't realise it was an argument used as an output. That makes sense.